### DISCUSSION TITLE
Use of uvm_sequence_item __eq__ method overload to compare results

### DISCUSSION THREAD

**Discussion by powelljd** [raysalemi/Python4RTLVerification#15] [Q&A] [answered]
I'm a newbie with regards to PyUVM so apologies if the question is silly... 

I'm struggling to find any use of the uvm_sequence_item `__eq__` method overload for comparing results. I've noticed that this is never used (neither the `__str__` overload) in any of the examples (or indeed the PyUVM TinyALU example). 

Comparisons seem to always be done with explicit tuples within the Scoreboard. Am I missing something?

**Comment by raysalemi** [comment]
It is Pythonic to be able to use `==` but one can also call the `compare()`
function.

**Reply by powelljd** [reply]
Understood, are there any examples where the uvm_sequence_item is directly used for a scoreboard check, either via the pythonic '==' or compare()? 


I've noticed Scoreboard check_phase throughout the examples never use uvm_sequence_item instances. I would have expected to see an instance of AluSeqItem used for comparison somewhere, for example: https://github.com/raysalemi/Python4RTLVerification/blob/b0897e4603c1e7417e5f30984ad59217f13ce017/41_Fibonacci_testbench_7.1/testbench.py#L148

If 'actual_result == predicted_result' were instances of AluSeqItem, but they are instead simply integers. Is this simply due to the example only having one variable to check (result)?

**Comment by raysalemi** [comment, marked as answer]
Yup. It was a simpler example if I didn’t add the *eq* to the example.

https://github.com/raysalemi/pyuvm/blob/master/tests/pytests/test_05_base_classes.py
shows a simple example:

class my_object(uvm_object):

    def __init__(self, name=""):

        super().__init__(name)

        self.val = 5


    def __eq__(self, other):

        if type(other) is type(self):

            return self.val == other.val


    def __str__(self):

        return "Hello"


    def do_copy(self, other):

        self.val = other.val


def test_create():

    """

    5.3.5 This needs to be further implemented to include the factory

    """

    mo = my_object("first")

    mo2 = mo.create("second")

    assert mo == mo2

**Reply by powelljd** [reply]
Great, thanks for confirming this. Great book also :).

**Reply by raysalemi** [reply]
Thanks!
